src/vs/workbench/contrib/terminal/common/terminalStorageKeys.ts

```diff
@@ -10,5 +10,11 @@ export const enum TerminalStorageKeys {
 	TabsListWidthVertical = 'tabs-list-width-vertical',
 	EnvironmentVariableCollections = 'terminal.integrated.environmentVariableCollections',
 	TerminalBufferState = 'terminal.integrated.bufferState',
-	TerminalLayoutInfo = 'terminal.integrated.layoutInfo'
+	TerminalLayoutInfo = 'terminal.integrated.layoutInfo',
+	PinnedRecentCommandsZsh = 'terminal.pinnedRecentCommandsZsh',
+	PinnedRecentCommandsBash = 'terminal.pinnedRecentCommandsBash',
+	PinnedRecentCommandsFish = 'terminal.pinnedRecentCommandsFish',
+	PinnedRecentCommandsPwsh = 'terminal.pinnedRecentCommandsPwsh',
+	PinnedRecentCommandsGitBash = 'terminal.pinnedRecentCommandsGitBash',
+	PinnedRecentCommandsPwshWindows = 'terminal.pinnedRecentCommandsPwshWindows'
```

Comment:
Should this instead use `${TerminalStorageKeys.PinnedRecentCommandsPrefix}.${this._shellType}`? There are only a limited number of shell types so it should be fine to do it that way. I see `PosixShellType.pwsh` and `WindowsShellType.pwsh` are both `"pwsh"`, but that's probably fine since I don't believe command history is synced.